Backend/routes/notifications.js

```diff
@@ -47,12 +46,40 @@ router.get("/notifications", verifyToken, async (req, res) => {
       orderBy: { createdAt: "desc" },
     });
 
-    const formattedNotifications = notifications.map((notification) => ({
-      ...notification,
-      timeAgo: formatDistanceToNow(new Date(notification.createdAt), {
-        addSuffix: true,
-      }),
-    }));
+    const groupedNotifications = notifications.reduce((acc, notif) => {
+      const key = `${notif.type}-${notif.listingId || ""}`;
+      if (!acc[key]) {
+        acc[key] = [];
+      }
+      acc[key].push(notif);
+      return acc;
+    }, {});
+
+    const formattedNotifications = Object.values(groupedNotifications).map(
+      (group) => {
+        if (group.length > 1 && group[0].type === "LIKE") {
+          const users = group.map((notif) => notif.usernameTarget).slice(0, 3);
+          const othersCount = group.length - users.length;
+          const content = `${users.join(", ")}${
+            othersCount > 0 ? `and ${othersCount} others` : ""
+          } liked your listing.`;
+          return {
+            ...group[0],
+            content,
+            timeAgo: formatDistanceToNow(new Date(group[0].createdAt), {
```

Comment:
will this first notif in group always be the one that was last created?

Reply:
Not necessarily, sometimes it is but I should definitely make it to be the most recently created all the time